src/js/components/modals/ServiceFormModal.js

```diff
@@ -0,0 +1,240 @@
+import Ace from 'react-ace';
+import mixin from 'reactjs-mixin';
+import {Modal} from 'reactjs-components';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import 'brace/mode/json';
+import 'brace/theme/monokai';
+import 'brace/ext/language_tools';
+
+import MarathonStore from '../../stores/MarathonStore';
+import SchemaForm from '../SchemaForm';
+import Service from '../../structs/Service';
+import ServiceUtil from '../../utils/ServiceUtil';
+import ServiceSchema from '../../constants/ServiceSchema';
+import ToggleButton from '../ToggleButton';
+
+const METHODS_TO_BIND = [
+  'getAdvancedSubmit',
+  'handleCancel',
+  'handleClearError',
+  'handleJSONChange',
+  'handleJSONToggle',
+  'handleSubmit',
+  'onMarathonStoreServiceCreateError',
+  'onMarathonStoreServiceCreateSuccess'
+];
+
+class ServiceFormModal extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    let model =
+      ServiceUtil.createFormModelFromSchema(ServiceSchema);
+    this.state = {
+      app: JSON.stringify({id:'', cmd:''}, null, 2),
+      errorMessage: null,
+      jsonMode: false,
+      model,
+      service: ServiceUtil.createServiceFromFormModel(model)
+    };
+
+    this.store_listeners = [
+      {
+        name: 'marathon',
+        events: ['serviceCreateError', 'serviceCreateSuccess']
+      }
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    super.componentWillReceiveProps(...arguments);
+    if (!this.props.open && nextProps.open) {
+      this.resetState();
+    }
+  }
+
+  resetState() {
+    let model = ServiceUtil.createFormModelFromSchema(ServiceSchema);
+    this.setState({
+      app: '',
+      errorMessage: null,
+      jsonMode: false,
+      model,
+      service: ServiceUtil.createServiceFromFormModel(model)
+    });
+  }
+
+  handleClearError() {
+    this.setState({
+      errorMessage: null
+    });
+  }
+
+  handleJSONChange(app) {
+    let {service} = this.state;
+    try {
+      service = new Service(JSON.parse(app));
+    } catch (e) {
+
+    }
+    this.setState({app, service})
+  }
+
+  handleJSONToggle() {
+    let nextState = {};
+    if (!this.state.jsonMode) {
+      let {model} = this.triggerAdvancedSubmit();
```

Comment:
The triggerAdvancedSubmit returns the data from the form including validation, model and definition of the form. This is needed to get the data from outside of the component. However the naming I must confess isn't good. I removed the Advanced part and the getter function for the trigger is called `getTriggerSubmit` instead of `getAdvancedSubmit`.